### DISCUSSION TITLE
[Question]  Bundle lib with Monaco component

### DISCUSSION THREAD

**Discussion by NicoLaval** [microsoft/monaco-editor#3698] [Q&A]
### Reproducible in vscode.dev or in VS Code Desktop?
* [x]  Not reproducible in [vscode.dev](https://vscode.dev/) or VS Code Desktop

### Reproducible in the monaco editor playground?
* [x]  Not reproducible in [the monaco editor playground](https://microsoft.github.io/monaco-editor/playground.html)

### Monaco Editor Playground Code
No response

### Actual Behavior
We are trying to ship a library that bundles monaco editor and [VTL Editor](https://github.com/eurostat/vtl-editor) using monaco webpack plugin.

As long as we do not bundle the lib, everything works as expected. See for example our [Storybook stories](https://eurostat.github.io/vtl-editor/index.html?path=/story/editor-vtl-2-0--default) ( [code here](https://github.com/eurostat/vtl-editor/blob/master/src/stories/0-VTL.stories.tsx) ).  If you go on and type full, intelisense will suggest full_join
However, when we try to use our [npm module](https://www.npmjs.com/package/@eurostat/vtl-editor) in a CRA, intelisense doesn’t work.

### Expected Behavior
When we bundle, deploy on NPM, and import the wrapped component into a client app, it fails. Workers don't seem to be found.

We have seen that we need ton configure the PublicPath of the plugin starting from webpack v5, but filling it with '', '/', ASSET_PATH, ... don't fix the problem.

Any idea?

### Additional Context
@EmmanuelDemey @garronej

**Comment by hediet** [comment]
> However, when we try to use our npm module in a CRA, intelisense doesn’t work.

Do you use the monaco-webpack plugin in the CRA?
It sets up the webworker entry points.

**Comment by NicoLaval** [comment]
I do, [here](https://github.com/eurostat/vtl-editor/blob/master/webpack.config.js#L32)
